### PR TITLE
fix(earn): Fix Earn Deposit bottom sheet bug on Android

### PR DESCRIPTION
**Pull request by jophish**
### Description

For [ACT-1398](https://linear.app/valora/issue/ACT-1398/android-pin-modal-display-obscured-earn-review-bottom-sheet). Fixes an issue where the deposit bottom sheet was obscuring the PIN entry screen.

### Test plan

Unit and manual tested, see video below:

https://github.com/user-attachments/assets/27140fb2-3ad5-4363-b458-7a4e8a7f1bb1

### Related issues

- Fixes #[issue number here]

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [ ] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

**Comment by MuckT** [comment]
@jophish is it possible to leave the bottom sheet displayed after the pin input and close prior or shortly after the navigation to home?

**Comment by satish-ravi** [comment]
+1 to Tom's comment, this means that if the user doesn't press continue again, they'll never know a tx was submitted right?

**Comment by jophish** [comment]
@satish-ravi @MuckT I've been trying to get this to work in various ways, but I just can't seem to figure out a way to get this to work reliably unfortunately. The PIN entry always seems to dismiss the bottom sheet screen.

**Comment by MuckT** [comment]
@jophish, it appears that on failed deposits, the bottom sheet does not dismiss at all, leaving the user on the tab's home screen with the bottom sheet displayed.

**Comment by jophish** [comment]
@MuckT What kind of failed deposit? I artificially induced an error during the deposit process (after the prompt for the user's PIN), and I landed back on the Enter Amount screen with no bottom sheet displayed.

File: src/earn/EarnEnterAmount.tsx
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:388]: this is just to make TS happy? is the `tokenAmount` check required? seems like there's already a check at the start of the function

**Review by satish-ravi — COMMENTED** [review]
Noticing 2 issues:
1. If a PIN is required, once the PIN screen opens the bottom sheet dismisses, and if the PIN is successfully entered, a tx is submitted on the background but the user is back on the EnterAmount screen with no feedback. Clicking continue is not the worst, as it opens the bottom sheet with the buttons disabled. But maybe we can make the "Continue" button a spinner if a tx is being submitted?
2. If a PIN is not required, the bottom sheet never dismisses even after the tx is submitted successfully. I can see the screen being navigated to home in the background. See attached video

https://github.com/user-attachments/assets/72b4ef03-4ffd-4fad-ba69-11ab2f2f3d22

**Comment by MuckT** [comment]
@jophish, the second issue @satish-ravi mentioned, where the bottom sheet is not dismissed if no PIN is requested, appears to introduce a new bug in iOS's happy path. Would it be better to close out ACT-1398, request new designs for a confirmation screen, and create a new ticket for that work?

**Comment by satish-ravi** [comment]
I just realized 1 is something we already discussed 😅 , but what about this idea. Regardless of whether pin is required or not, when "Continue" is pressed on the bottom sheet, we dismiss the bottom sheet, and add a loading spinner to the "Continue" button on enter amount. I think its a workaround that solves both 1 and 2. But maybe not the greatest of experiences.

**Comment by jophish** [comment]
@satish-ravi I think I like this idea; I'll make this change for the interim, but in addition I think it's worth requesting new designs for this anyways, since this isn't the most ideal solution imo.

**Comment by jophish** [comment]
Updated to reflect the discussion above, see new video here:

https://github.com/user-attachments/assets/27140fb2-3ad5-4363-b458-7a4e8a7f1bb1

File: src/earn/EarnEnterAmount.tsx
Comment (jophish) [reply src/earn/EarnEnterAmount.tsx:392]: Oh good call, yeah I think this can be removed

File: src/earn/EarnEnterAmount.test.tsx
Comment (MuckT) [inline src/earn/EarnEnterAmount.test.tsx:284]: nit if true: should these tests be more specific in their asserts?

**Review by MuckT — APPROVED** [review]
Thanks for taking the time to get this working well on Android an iOS testing locally, my concerns with the happy path on both Android and iOS have been resolved.

File: src/earn/EarnEnterAmount.tsx
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:105]: think this screen is reused for withdraw too, should there be an additional check? probably not required, unless the redux state is corrupted

**Review by satish-ravi — APPROVED** [review]

File: src/earn/EarnEnterAmount.tsx
Comment (jophish) [reply src/earn/EarnEnterAmount.tsx:105]: I don't think this is strictly necessary, since pressing "continue" for a withdrawal actually takes you to another confirmation screen; I can leave a comment however.

**Pull request merged**
